### DISCUSSION TITLE
2024 November Release 1 (v4.1.0)

### DISCUSSION THREAD

**Discussion by donaldzou** [WGDashboard/WGDashboard#459] [General]
# 📣 What's New: v4.1.0

## 🎉 New Features
- **Multi-Language Support**: Now WGDashboard support the following languages on its user interface, big thanks to our user's contribution!
  - **Chinese Traditional**
  - **Chinese Simplified**
  - **Czech** (@jursed)
  - **Dutch** (@DaanSelen)
  - **English**
  - **German** (@orangeferdi)
  - **Italian** (@3vis97)
  - **Russian** (Pixnet)
  - **Ukrainian** (@shuricksumy)

> If you would like to contribute, please follow the instructions on [Localization of WGDashboard](https://github.com/donaldzou/WGDashboard/issues/397). Thanks in advance!

- **Backup & Restore WireGuard Configurations**: Now you can back up your configurations, restore it after a change made to the configuration. **You can also restore it even after deletion.**
- **Delete & Rename WireGuard Configuration:** Now you can delete and rename configuration within WGDashboard
- **Toggle WireGuard Configuration After Startup:** Now you can set WireGuard configurations to be turned on after starting WGDashboard in **Settings**
- **Delete & Download Peers in bulk**
- **Frontend Display of Peer's Configuration File**
- **Added Support on AlmaLinux and Pi OS**
- **Added OpenStreetMap on Ping and Traceroute Tool**

## 🛠️ Some Adjustments
- **Updated Docker configuration**
- **Updates on API endpoints**
- UI Adjustments
- Added version number in navbar
- Added WGDashboard host and port settings
- Added peer delete confirmation
- Added domain support in DNS field for peers

## 🧐 Bugs Fixed
- Mobile UI issues in #353
- Removed WireGuard configuration error alert from Gunicorn start in #328
- Sometimes restrict peer might not be success in #357
- Weird SQLite error causing WGDashboard to crash in #366

## 🗂️ User Guides
Will continue to finish the [](User-Guides.md) sections

## 🥘 Experimental Features
- **Cross-Server Access**: Now you can access other servers that installed `v4` of WGDashboard through API key.
- **Desktop App**: Thanks to **Cross-Server Access**, you can now download an ElectronJS based desktop app of WGDashboard, and use that to access WGDashboard on different servers.
> For more information, please visit [🥘 Experimental Features](https://donaldzou.github.io/WGDashboard-Documentation/experimental-features.html)

### Solution 1 (v4.0+)

```shell
sudo ./wgd.sh update
```

### Solution 2

> If **Solution 1** above is not working for you...

1. Change your directory to `wgdashboard`

    ```shell
    cd wgdashboard/src
    ```

2. Update the dashboard
    ```shell
    git pull https://github.com/donaldzou/WGDashboard.git --force
    ```

3. Install

   ```shell
   sudo ./wgd.sh install
   ```

<hr /><em>This discussion was created from the release <a href='https://github.com/donaldzou/WGDashboard/releases/tag/v4.1.0'>2024 November Release 1 (v4.1.0)</a>.</em>

**Comment by shuricksumy** [comment]
Great job. @donaldzou 

Thanks a lot for the new release. 
It has a lot of improvements and new functionality.

**Comment by DaanSelen** [comment]
Lets go @donaldzou ! It keeps getting better! 🥳🥳❤️❤️

**Comment by NovianPutera** [comment]
Nice Update. Thank you very much 🎊🎊
